src/reachy2_sdk/parts/mobile_base.py

```diff
@@ -0,0 +1,284 @@
+
+        req = TargetDirectionCommand(
+            direction=DirectionVector(
+                x=FloatValue(value=x_vel),
+                y=FloatValue(value=y_vel),
+                theta=FloatValue(value=deg2rad(rot_vel)),
+            )
+        )
+        self._mobility_stub.SendDirection(req)
+
+    def goto(
+        self,
+        x: float,
+        y: float,
+        theta: float,
+        timeout: Optional[float] = None,
+        tolerance: Dict[str, float] = {"delta_x": 0.1, "delta_y": 0.1, "delta_theta": 15, "distance": 0.1},
+    ) -> None:
+        """Send target position. x, y are in meters and theta is in degree.
+
+        (x, y) will define the position of the mobile base in cartesian space
+        and theta its orientation. The zero position is set when the mobile base is
+        started or if the  reset_odometry method is called.
+        A timeout in seconds is defined so that the mobile base does get stuck in a go
+        to call.
+        The tolerance represents the margin along x, y and theta for which we consider
+        that the mobile base has arrived its goal.
+        """
+        if self.is_off():
+            raise RuntimeError(("Mobile base is off. Goto not sent."))
+
+        exc_queue: Queue[Exception] = Queue()
+
+        if not timeout:
+            # We consider that the max velocity for the mobile base is 0.5 m/s
+            # timeout is 2*_max_xy_goto / max velocity
+            timeout = 2 * self._max_xy_goto / 0.5
+
+        def _wrapped_goto() -> None:
+            try:
+                asyncio.run(
+                    self._goto_async(
+                        x=x,
+                        y=y,
+                        theta=theta,
+                        timeout=timeout,
+                        tolerance=tolerance,
+                    ),
+                )
+            except Exception as e:
+                exc_queue.put(e)
+
+        with ThreadPoolExecutor() as exec:
+            exec.submit(_wrapped_goto)
+        if not exc_queue.empty():
+            raise exc_queue.get()
+
+    async def _goto_async(
+        self,
+        x: float,
+        y: float,
+        theta: float,
+        timeout: float,
+        tolerance: Dict[str, float] = {"delta_x": 0.1, "delta_y": 0.1, "delta_theta": 15, "distance": 0.1},
+    ) -> None:
+        """Async version of the goto method."""
+        for pos, value in {"x": x, "y": y}.items():
+            if abs(value) > self._max_xy_goto:
+                raise ValueError(f"The asbolute value of {pos} should not be more than {self._max_xy_goto}!")
+
+        req = GoToVector(
+            x_goal=FloatValue(value=x),
+            y_goal=FloatValue(value=y),
+            theta_goal=FloatValue(value=deg2rad(theta)),
+        )
+        self._drive_mode = "go_to"
+        self._mobility_stub.SendGoTo(req)
+
+        tic = time.time()
+        arrived: bool
```

Comment:
initialize arrived with a value

Reply:
This while means that we are waiting for the sendgoto to complete right? Maybe we can add a comment or move it to a function so it's more explicit